src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -34,9 +35,9 @@ public class SchemaInitializer : IHostedService
         private readonly ISqlConnectionStringProvider _sqlConnectionStringProvider;
         private readonly IMediator _mediator;
 
-        public SchemaInitializer(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ISqlConnectionFactory sqlConnectionFactory, ISqlConnectionStringProvider sqlConnectionStringProvider, IMediator mediator, ILogger<SchemaInitializer> logger)
+        public SchemaInitializer(IOptions<SqlServerDataStoreConfiguration> sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ISqlConnectionFactory sqlConnectionFactory, ISqlConnectionStringProvider sqlConnectionStringProvider, IMediator mediator, ILogger<SchemaInitializer> logger)
         {
-            _sqlServerDataStoreConfiguration = EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+            _sqlServerDataStoreConfiguration = EnsureArg.IsNotNull(sqlServerDataStoreConfiguration?.Value, nameof(sqlServerDataStoreConfiguration));
```

Comment:
I though you need GetCurrentSchemaVersionAsync() out of here so that you can use it out of IHostedService

Reply:
You're right, I just left that out of the initial PR haha. I have added a new "read-only" interface `IReadOnlySchemaManagerDataStore` that allows users to query the version without the management hosted service.

Reply:
Thanks. I am not sure we have a variation of the implementation here and one in SchemaManagerDataStore

Reply:
I removed the implementation from `SchemaInitializer` and replaced it with `SchemaManagerDataStore`